x-pack/plugins/lens/public/state_management/lens_slice.ts

```diff
@@ -93,16 +93,20 @@ export const getPreloadedState = ({
     };
   }
 
+  const query = !initialContext
```

Comment:
It feels like it ought to be a separate function to unroll all this cascading stuff into something more readable.
It started with a ternary check, it is becoming some sort of living creature....